### ISSUE TITLE
"Cannot hook to Dolphin, the process is running, but no emulation has been started" error

### ISSUE THREAD

**Issue by calvinpelletier** [aldelaro5/Dolphin-memory-engine#35]
I'm getting "Cannot hook to Dolphin, the process is running, but no emulation has been started" even though an emulation is running. Any thoughts? I'm using Dolphin 5.0.

**Comment by aldelaro5** [comment]
I just want to comment on this saying you are not the only person that reported issues with the base address since. The tool I linked above is a good one to debug this because you can check which pages looks similar, the issue is more that it's complicated to use for end users so I will be checking if I should simply log the pages mapped by Dolphin and display them for debugging purposes so that way, I can see why no pages were chosen for the heuristics and Dolphin can report the TRUE address using the logger and the instructions above.

But yeah, so far, this is a mystery.

**Comment by moomath** [comment]
I am running this on linux and it took a while to set up but I was wondering if any information has been released on what to do? Is there just no solution yet? Thank you.

**Comment by aldelaro5** [comment]
Still unsure how to replicate this unfortunately. It seems to be very inconsistent across users.

**Comment by aldelaro5** [comment]
@moomath oh, have you tried this from the readme?

Troubleshouting

On Linux, the program may require additional kernel permissions to be able to read and write memory to external processes (which is required to read and write the memory of Dolphin). If nothing happens to Dolphin, but the program frequently unhooks itself, the program is missing the required permissions. Grant these permissions by running the following command as root:

setcap cap_sys_ptrace=eip DME

Where DME is the path of the Dolphin Memory Engine executable. This should fix the permission problem for future executions.

**Comment by moomath** [comment]
@aldelaro5 yes, i was able to get it to work with your help, haha, since we were talking on discord just now.

**Comment by csunday95** [comment]
@aldelaro5 I am experiencing the same "the process is running, but no emulation has been started" issue despite a GC game being emulated. My setup is

Dolphin 5.0 Stable

Ubuntu 18.04.3 LTS

Dolphin-memory-engine Beta 0.5 (built from source)

I have performed the setcap command outlined above, but it had no effect.

**Comment by Gamebuster19901** [comment]
I'm also experiencing this issue, where it says "the process is running, but no emulation has been started" despite a wii game being emulated.

My setup is:

Dolphin 5.0 Stable

Ubuntu 18.04.4 LTS

Dolphin-memory-engine Beta 0.5.

I have also executed the setcap command above as the root user with no effect.

I have also tried running memory engine as sudo with no effect.

**Comment by Gamebuster19901** [comment]
I believe I have found the issue.

The logic here seems incorrect:

https://github.com/aldelaro5/Dolphin-memory-engine/blob/9e80462b7aa7a2b002843dd3cd9b3e2f45d0c9e0/Source/DolphinProcess/Linux/LinuxDolphinProcess.cpp#L29-L30

It assumes the memory addresses are always a length of 12. On my system, the memory address has a length of 10.

```
2300000000-2302000000 rw-s 00000000 00:18 904                            /dev/shm/dolphinmem.0 (deleted)
2310000000-2314000000 rw-s 02040000 00:18 904                            /dev/shm/dolphinmem.0 (deleted)
2500000000-2502000000 rw-s 00000000 00:18 904                            /dev/shm/dolphinmem.0 (deleted)
2580000000-2582000000 rw-s 00000000 00:18 904                            /dev/shm/dolphinmem.0 (deleted)
2590000000-2594000000 rw-s 02040000 00:18 904                            /dev/shm/dolphinmem.0 (deleted)
25c0000000-25c2000000 rw-s 00000000 00:18 904                            /dev/shm/dolphinmem.0 (deleted)
25d0000000-25d4000000 rw-s 02040000 00:18 904                            /dev/shm/dolphinmem.0 (deleted)
25e0000000-25e0040000 rw-s 02000000 00:18 904                            /dev/shm/dolphinmem.0 (deleted)
```

So when it goes to read characters 0 through 12 of the first entry, it will get `2300000000-2` instead of `2300000000`.

Instead of assuming the length is 12, keep reading until you reach the first `-` character and then you will know the length.

**Comment by aldelaro5** [comment]
Interesting, I might consider pushing a hotfix for this one because it's been reported MANY times and it is blocking.

If I do however, it will take a while as you can probably tell this repos is kind of dead at the moment for many reasons one of them being I had plans to port this to dolphin, but other things came up and now it feels awkward to work on this. I might make an exception on this one, but in the mean time, if you submit a PR, I could make a hotfix release.

**Comment by Gamebuster19901** [comment]
I've never coded in c++ before but I'll see what I can do.

**Comment by Gamebuster19901** [comment]
Well, I was able to get to to hook, but it immediately unhooks right afterwards. I'm unable to figure out why.

My fork is here if anyone wants to give it a go: https://github.com/Gamebuster19901/Dolphin-memory-engine

**Comment by aldelaro5** [comment]
> Well, I was able to get to to hook, but it immediately unhooks right afterwards. I'm unable to figure out why.
> 
> My fork is here if anyone wants to give it a go: https://github.com/Gamebuster19901/Dolphin-memory-engine

This happens on my end too, seems like a different issue, I will be testing now.

**Comment by aldelaro5** [comment]
Fixed completely by 5d8fd7fc6865fcb066ae5ac79ddcc22b8cd42cb4

Just remember the setcap command though, on my system, it seems to be more aggressive with it than before when I tested it
